applications/FreeSurfaceApplication/custom_utilities/edgebased_gid_io.h

```diff
@@ -576,6 +568,8 @@ class EdgebasedGidMeshContainer
     const char* mMeshTitle;
 };//class EdgebasedGidMeshContainer
 
+KRATOS_API_EXTERN template class KRATOS_API(FREE_SURFACE_APPLICATION) GidIO<EdgebasedGidGaussPointsContainer,EdgebasedGidMeshContainer>;
```

Comment:
@roigcarlo @joaquinirazabal just for me to know,  can you clarify why this is needed? If you do the explicit instantiation in here why it is done again in the new cpp?

Reply:
The one in the cpp is the explicit instantiation, the one in the h is the declaration with the classes from the FreeSurface. Both are needed as the base clase is templated and derived from the core

Reply:
Can you please copy&paste this as an inline comment in the code?